var/spack/repos/builtin/packages/lammps/package.py

```diff
@@ -220,6 +220,10 @@ def cmake_args(self):
             cxx_flags = '-Ofast -mfma -fvectorize -funroll-loops'
             args.append(self.define('CMAKE_CXX_FLAGS_RELEASE', cxx_flags))
 
+        if spec.satisfies('%intel') and spec.satisfies('target=zen:'):
+            cmake_tune_flags="-march=core-avx2 -fma"
+            args.append(self.define('CMAKE_TUNE_FLAGS', cmake_tune_flags))
```

Comment:
We shouldn't add any compiler flag that might change which instructions are emitted in `package.py`, since it might conflict with flags injected by archspec. What is the issue we are trying to solve? Currently `archspec` injects `-march=core-avx2` for this combination. Should `-fma` be added in general when we try to optimize for ZenX?

Reply:
The issue I am trying to solve: If `CMAKE_TUNE_FLAGS` is unset `-xHost` is inserted to the compile line by Cmake. This is sub-optimal for Intel compiler on Zen as it will not issue AVX instructions.

My experience has shown `-march=core-avx2 -fma` to be the best combination of flags for Intel compielr on Zen. I would add it in general. I have not checked any assembly code though.